component/prometheus/scrape/scrape_test.go

```diff
@@ -9,12 +9,71 @@ import (
 	"github.com/grafana/agent/component"
 	"github.com/grafana/agent/component/prometheus"
 	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/pkg/river"
 	prometheus_client "github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/storage"
 	"github.com/stretchr/testify/require"
 )
 
+func TestRiverConfig(t *testing.T) {
+	var exampleRiverConfig = `
+	targets         = [{ "target1" = "target1" }]
+	forward_to      = []
+	scrape_interval = "10s"
+	job_name        = "local-flow"
+
+	bearer_token = "token"
+	bearer_token_file = "/path/to/file.token"
+	proxy_url = "http://0.0.0.0:11111"
+	follow_redirects = true
+	enable_http2 = true
+
+	basic_auth {
+		username = "user"
+		password = "password"
+		password_file = "/path/to/file.password"
+	}
+
+	authorization {
+		type = "Bearer"
+		credentials = "credential"
+		credentials_file = "/path/to/file.credentials"
+	}
+
+	oauth2 {
+		client_id = "client_id"
+		client_secret = "client_secret"
+		client_secret_file = "/path/to/file.oath2"
+		scopes = ["scope1", "scope2"]
+		token_url = "token_url"
+		endpoint_params = {"param1" = "value1", "param2" = "value2"}
+		proxy_url = "http://0.0.0.0:11111"
+		tls_config = {
```

Comment:
this is an attribute here and block below. That doesn't feel right. should I update to block for both?

Reply:
It seems like it was a mistake for this to be a block, and we probably document it as if it's supposed to be a block. Changing it makes sense.

Reply:
Changing this would technically be a breaking change, but if our documentation says that it's supposed to be block, then we can list it as a bug fix IMO.

Reply:
see my blind stab at the changelog. I am wondering if we want a blanket statement on these acknowledging breaking changes to config in general or something more specific for each component.

Reply:
updated and documented as a bug fix for all impacted components